web/src/recent_view_ui.ts

```diff
@@ -662,6 +662,7 @@ function format_conversation(conversation_data: ConversationData): ConversationC
         const recipient_id = last_msg.recipient_id;
         const pm_url = last_msg.pm_with_url;
         const is_group = last_msg.display_recipient.length > 2;
+        const has_unread_mention = unread.get_counts().mention_dict.get(user_ids_string);
```

Comment:
This has the same major performance bug as the inbox version of this change. Again, we have `message_to_conversation_unread_count` above with access to unread data.